src/tokenizer/lexer.rb

```diff
@@ -197,27 +229,30 @@ def else?(chunk)
     end
 
     def comp_1?(chunk)
-      chunk =~ /.+が$/ && variable?(chunk.chomp('が'))
+      chunk =~ /.+が$/ && @context.inside_if_condition?
     end
 
     def comp_2?(chunk)
-      variable?(chunk) && question?(@reader.peek_next_chunk)
+      !chunk.empty? && question?(@reader.peek_next_chunk)
     end
 
     def comp_2_to?(chunk)
-      chunk =~ /.+と$/ && variable?(chunk.chomp('と'))
+      chunk =~ /.+と$/ && begin
+        next_chunk = @reader.peek_next_chunk
+        comp_3_eq?(next_chunk) || comp_3_neq?(next_chunk)
+      end
     end
 
     def comp_2_yori?(chunk)
-      chunk =~ /.+より$/ && variable?(chunk.chomp('より'))
+      chunk =~ /.+より$/
```

Comment:
Should add note to token_sequence, just in case...

If `COMP_2` is checked _after_ `COMP_2_YORI` (or anything else), then

```
あよりは 1
もし 1が あより？ ならば
```
will be parsed as a `COMP_2_YORI` instead of a `COMP_2` and will fail when it hits the question mark.

We can add a `peek_next_chunk` to the `COMP_2_*` matchers instead, but just adding a note about order is probably enough for now.